### DISCUSSION TITLE
[FR - Craft 5] Relate to Matrix Block Entries using the Entries field

### DISCUSSION THREAD

**Discussion by steverowling** [craftcms/cms#13794] [Ideas]
Now that Matrix Blocks are entries in Craft 5, it would be great to be able to use Matrix Block Entries as relation targets in an Entries field, so you could link directly to a Matrix Block Entry in the same way that you can relate to an Entry in a section.

**Comment by brandonkelly** [comment]
I just made this possible in a roundabout way, by adding a new “Matrix field” condition rule to entry conditions (711c8de631c98f14bc9b474dccfd20dd17642c78). With that change, you could:

1. Create a new custom entry source that pulls in entries from a particular Matrix field via the new condition rule. (You can disable the source for all user groups so only admins actually know it exists.)
2. Select your custom source from an Entries field’s settings.

That will just list all of the nested entries together though, without providing any context, not super intuitive. I suppose ideally it would be nice if Entries fields could be configured to show entries from a given section(s), and let you navigate down into their Matrix fields from there to make your actual selection. Sortof similar to asset folder navigation.

**Reply by thanasispsarros** [reply]
> I just made this possible in a roundabout way, by adding a new “Matrix field” condition rule to entry conditions ([711c8de](https://github.com/craftcms/cms/commit/711c8de631c98f14bc9b474dccfd20dd17642c78)). With that change, you could:
> 
> 1. Create a new custom entry source that pulls in entries from a particular Matrix field via the new condition rule. (You can disable the source for all user groups so only admins actually know it exists.)
> 2. Select your custom source from an Entries field’s settings.
> 
> That will just list all of the nested entries together though, without providing any context, not super intuitive. I suppose ideally it would be nice if Entries fields could be configured to show entries from a given section(s), and let you navigate down into their Matrix fields from there to make your actual selection. Sortof similar to asset folder navigation.

Hi @brandonkelly! I _think_ I had tested this some time after your original comment, but has there been any change that prevents this from working in current versions? When I follow the steps above the entry selection modal is empty for any non-admin users. The only way I can get it to work is if, in the Entries field's settings, I select an active source (or multiple sources) and then add the Matrix field condition.

**Comment by steverowling** [comment]
That's a great start, thanks! For my use case, the relations would be made programmatically, so the real benefit would be being able to navigate to the related nested entry directly in a slide out. However, I can definitely see the potential and value in being able to do this as well:

> I suppose ideally it would be nice if Entries fields could be configured to show entries from a given section(s), and let you navigate down into their Matrix fields from there to make your actual selection. Sort of similar to asset folder navigation.

Would it be possible to be able to show nested entry context in an entry chip or card? Perhaps by including the title of the parent entry? Or a combination of title/matrix field name.

**Reply by brandonkelly** [reply]
Maybe, but I’d be concerned about performance. Can you post a new discussion about that?
